### DISCUSSION TITLE
Down migration missing in v4

### DISCUSSION THREAD

**Discussion by abenerd** [spatie/laravel-tags#309] [General]
I noticed that in v4 the down migration has been removed and I was wondering what the reason for this removal was.
Of course I can easily ad it back myself, but I'm interested in the reason for this this removal.
Thanks in advance.

**Comment by pardalsalcap** [comment]
Hi!
I was writing a new discussion but found this one.
Maybe is intentional, but I think a down method is missing in the migration file.

Scenario:
On a fresh installation, you may need to refresh the database, so if the database has the two tables that the package requires, an error like this pops up:

SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'tags' already exists 

A down method like this solves this situation

public function down()
    {
        Schema::dropIfExists("taggables");
        Schema::dropIfExists("tags");
    }

Thanks for your time and your amazing packages and courses.

**Comment by freekmurze** [comment]
Thanks for your suggestion. We never use down migrations, so none of our packages contain one. If you need to migrate down, feel free to add a down migration to your project.
